### DISCUSSION TITLE
How to display persona card on hover with custom result source (merge of user and list items of SharePoint list) v4.9.0

### DISCUSSION THREAD

**Discussion by DominikLeitgeb** [microsoft-search/pnp-modern-search#3094] [Q&A]
Tested with PnP Modern Search v4.7.0 and 4.9.0

**Used Webparts:**

- Search Result
- Search Box
- Search Refiners

**Use case:** 

Merging SharePoint user items and list items of a SharePoint list with contact data such as name, phone number, mobile number and other additional data into one search experience. The contact data is edited in AD (onPrem) and synchronized into the list.

**Current procedure:**

1. A content type was created for the contacts and the necessary columns were assigned.
2. Created a result type for the contacts in the list and configured it similar to the "People" result type in SharePoint.
3. Created a search result source that combines the SPO users and the items from the contact list. The query looks like this: {searchTerms} AND (ContentClass=urn:content-class:SPSPeople OR ContentTypeId: "0x0100C2AE178EAF5023458908F62EA173218A*")
4. The search result source ID was copied to the result source column of the search result weboart and the "Persona Card on Hover" was enabled.

The search result webpart displays all items, unfortunately the Persona Card on Hover does not work for the User Items.
Trying to customize the Handlebars template and display the Persona Card only for the User Items did not work either.

How can the Persona Card be displayed for the User Items with the custom search result source in the search result?

**Expected behavior**
The Persona Cards should be displayed for the SharePoint User Items in the search result, because there is a PersonQuery for these items.
For the items from the contact list, the person cards would be nice, but are actual not mandatory.

**Screenshots**
![ResultWebpartConfig2](https://github.com/microsoft-search/pnp-modern-search/assets/87366596/adf0334b-9af9-499e-945b-0e2186186b17)
![ResultWebpartConfig1](https://github.com/microsoft-search/pnp-modern-search/assets/87366596/d04d8d33-e12b-4d9d-b392-bf059d14bdea)
![ResultTypeConfig](https://github.com/microsoft-search/pnp-modern-search/assets/87366596/513ec0dd-2451-4a47-86ba-5a5db6ce092a)
![ResultSourceConfig2](https://github.com/microsoft-search/pnp-modern-search/assets/87366596/56ebec41-a30e-4392-92f4-0cc5e33f6ec1)
![ResultSourceConfig1](https://github.com/microsoft-search/pnp-modern-search/assets/87366596/80b601a4-85e5-4469-8d9f-c4f3986de9b3)

**Config. File Content**

{
	"documentationLink": "https://microsoft-search.github.io/pnp-modern-search/",
	"selectedLayoutKey": "People",
	"resultTypes": [],
	"dataSourceProperties": {
		"queryTemplate": "{searchTerms}*",
		"enableQueryRules": false,
		"enableLocalization": false,
		"refinementFilters": "",
		"selectedProperties": [
			"AuthorOWSUSER",
			"ContactDepartmentOWSTEXT",
			"ContactFaxOWSTEXT",
			"ContactMobileOWSTEXT",
			"ContactPhoneOWSTEXT",
			"contentclass",
			"ContentTypeId",
			"Created",
			"CreatedBy",
			"DefaultEncodingURL",
			"FileType",
			"HitHighlightedProperties",
			"HitHighlightedSummary",
			"HtmlFileType",
			"JobTitle",
			"NormListID",
			"NormSiteID",
			"NormUniqueID",
			"NormWebID",
			"owstaxidmetadataalltagsinfo",
			"Path",
			"SiteLogo",
			"SiteTitle",
			"SPSiteUrl",
			"Title",
			"UniqueID",
			"UserName",
			"WorkPhone"
		],
		"resultSourceId": "a1b58505-b34c-4901-9795-a8da73214807",
		"hitHighlightedProperties": "",
		"trimDuplicates": false,
		"sortList": []
	},
	"queryTextSource": 1,
	"layoutProperties": {
		"peopleFields": [
			{
				"name": "Bild URL",
				"field": "imageUrl",
				"value": "/_layouts/15/userphoto.aspx?size=L&username={{getUserEmail (slot item @root.slots.UserEmail)}}",
				"useHandlebarsExpr": true,
				"supportHtml": false,
				"sortIdx": 1
			},
			{
				"name": "Primärer Text",
				"field": "primaryText",
				"value": "{{slot item @root.slots.UserDisplayName}}",
				"useHandlebarsExpr": true,
				"supportHtml": true,
				"sortIdx": 2
			},
			{
				"name": "Sekundärer Text",
				"field": "secondaryText",
				"value": "{{#contains ContentTypeId \"0x0100C2AE178EAF5023458908F62EA173218A\"}}\n    {{#if (slot item @root.slots.ContactDepartment)}}\n    {{slot item @root.slots.ContactDepartment}}\n    {{/if}}\n{{else}}\n    {{slot item @root.slots.JobTitle}}\n{{/contains}}\n",
				"useHandlebarsExpr": true,
				"supportHtml": true,
				"sortIdx": 3
			},
			{
				"name": "Tertiärer Text",
				"field": "tertiaryText",
				"value": "{{#contains ContentTypeId \"0x0100C2AE178EAF5023458908F62EA173218A\"}}\n    {{#if (slot item @root.slots.ContactPhone)}}\n    Phone: {{slot item @root.slots.ContactPhone}}\n    {{/if}}\n    {{#if (slot item @root.slots.ContactMobile)}}\n    Mobile: {{slot item @root.slots.ContactMobile}}\n    {{/if}}\n    {{#if (slot item @root.slots.ContactFax)}}\n    Fax: {{slot item @root.slots.ContactFax}}\n    {{/if}}\n{{else}}\n    {{getUserEmail (slot item @root.slots.UserEmail)}}\n{{/contains}}\n",
				"useHandlebarsExpr": true,
				"supportHtml": true,
				"sortIdx": 4
			},
			{
				"name": "Optionaler Text",
				"field": "optionalText",
				"value": "WorkPhone",
				"useHandlebarsExpr": false,
				"supportHtml": true,
				"sortIdx": 5
			}
		],
		"personaSize": 14,
		"showPersonaCard": true
	},
	"showSelectedFilters": false,
	"showResultsCount": true,
	"showBlankIfNoResult": false,
	"useMicrosoftGraphToolkit": false,
	"enableTelemetry": true,
	"itemSelectionProps": {
		"allowItemSelection": false,
		"selectionMode": "AsDataFilter",
		"allowMulti": false,
		"valuesOperator": "or"
	},
	"extensibilityLibraryConfiguration": [
		{
			"name": "Standard-Erweiterungsbibliothek",
			"enabled": true,
			"id": "dc4f961b-dbe0-44b4-982d-5776bf99d015"
		}
	],
	"queryModifierConfiguration": [],
	"queryModifierProperties": {
		"endWhenSuccessfull": false
	},
	"selectedVerticalKeys": [],
	"useVerticals": false,
	"paging": {
		"itemsCountPerPage": 6,
		"pagingRange": 5,
		"showPaging": true,
		"hideDisabled": true,
		"hideFirstLastPages": false,
		"hideNavigation": false,
		"useNextLinks": false
	},
	"adaptiveCardsHostConfig": "{\n\t\"fontFamily\": \"Segoe UI, Helvetica Neue, sans-serif\"\n}",
	"layoutRenderType": "Handlebars",
	"dataSourceKey": "SharePointSearch",
	"templateSlots": [
		{
			"slotName": "Title",
			"slotField": "Title",
			"sortIdx": 1
		},
		{
			"slotName": "Path",
			"slotField": "DefaultEncodingURL",
			"sortIdx": 2
		},
		{
			"slotName": "Summary",
			"slotField": "HitHighlightedSummary",
			"sortIdx": 3
		},
		{
			"slotName": "FileType",
			"slotField": "FileType",
			"sortIdx": 4
		},
		{
			"slotName": "PreviewImageUrl",
			"slotField": "AutoPreviewImageUrl",
			"sortIdx": 5
		},
		{
			"slotName": "PreviewUrl",
			"slotField": "AutoPreviewUrl",
			"sortIdx": 6
		},
		{
			"slotName": "Author",
			"slotField": "AuthorOWSUSER",
			"sortIdx": 7
		},
		{
			"slotName": "Tags",
			"slotField": "owstaxidmetadataalltagsinfo",
			"sortIdx": 8
		},
		{
			"slotName": "Date",
			"slotField": "Created",
			"sortIdx": 9
		},
		{
			"slotName": "SiteId",
			"slotField": "NormSiteID",
			"sortIdx": 10
		},
		{
			"slotName": "WebId",
			"slotField": "NormWebID",
			"sortIdx": 11
		},
		{
			"slotName": "ListId",
			"slotField": "NormListID",
			"sortIdx": 12
		},
		{
			"slotName": "ItemId",
			"slotField": "NormUniqueID",
			"sortIdx": 13
		},
		{
			"slotName": "IsFolder",
			"slotField": "ContentTypeId",
			"sortIdx": 14
		},
		{
			"slotName": "contentclass",
			"slotField": "contentclass",
			"sortIdx": 15
		},
		{
			"slotName": "PersonQuery",
			"slotField": "AADObjectID",
			"sortIdx": 16
		},
		{
			"slotName": "UserDisplayName",
			"slotField": "Title",
			"sortIdx": 17
		},
		{
			"slotName": "UserEmail",
			"slotField": "UserName",
			"sortIdx": 18
		},
		{
			"uniqueId": "e1fbdf79-26f4-482d-8e03-1150bdb9202a",
			"slotName": "ContactDepartment",
			"slotField": "ContactDepartmentOWSTEXT",
			"sortIdx": 19
		},
		{
			"uniqueId": "2fe98b81-5f94-440c-969f-c90f32eee115",
			"slotName": "ContactFax",
			"slotField": "ContactFaxOWSTEXT",
			"sortIdx": 20
		},
		{
			"uniqueId": "ae4fef34-f2ec-4cd3-a288-32708b75d4cb",
			"slotName": "ContactMobile",
			"slotField": "ContactMobileOWSTEXT",
			"sortIdx": 21
		},
		{
			"uniqueId": "fa0ef5de-c586-4164-b8b9-ea258b9d3bde",
			"slotName": "ContactPhone",
			"slotField": "ContactPhoneOWSTEXT",
			"sortIdx": 22
		},
		{
			"uniqueId": "5044f500-ff75-4a20-a805-ee1583b0d029",
			"slotName": "JobTitle",
			"slotField": "JobTitle",
			"sortIdx": 23
		}
	],
	"inlineTemplateContent": null,
	"externalTemplateUrl": "",
	"useInputQueryText": true,
	"queryText": {
		"__type": "DynamicProperty",
		"reference": {
			"_reference": "WebPart.544c1372-7e5a-49ec-8db6-812f76c375f2.bea38c17-6340-4598-adb6-034057725c1a:pnpSearchBoxWebPart",
			"_sourceId": "WebPart.544c1372-7e5a-49ec-8db6-812f76c375f2.bea38c17-6340-4598-adb6-034057725c1a",
			"_property": "pnpSearchBoxWebPart"
		}
	},
	"filtersData": {
		"__type": "DynamicProperty",
		"value": {}
	},
	"selectedItemFieldValue": {
		"__type": "DynamicProperty",
		"value": ""
	}
}

**Desktop:**
- Browser: Chrome
- OS Win 11 Enterprise v22H2

**Comment by wobba** [comment]
The SharePoint API no longer support merging user profiles and list items. You can try to use result blocks.. one for each, and then merge them with handlebars perhaps.

**Comment by DominikLeitgeb** [comment]
Thanks for the quick reply.

I am not sure how to create such result blocks to merge them into one handlebar.
Do you have a short description or any link for me for this specific case?
I would be very grateful for a short guidance.

**Reply by wobba** [reply]
If you have a query rule bringing in a result block of e.g. either the contacts or all the results (limited to 10 results), then enable query rules in the web part. Use the debug display template to look at the structure of the returned json. Might not be a built in function to join/concat two arrays, but you could loop each one.

```{{#each ```
